toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -440,6 +441,26 @@ func PopulateInstallRoot(installChroot *safechroot.Chroot, packagesToInstall []s
 		return
 	}
 
+	if isRootFS && (len(config.Users) > 0 || len(config.Groups) > 0) {
+		// Install shadow-utils package
```

Comment:
I'd prefer to see this check moved to the validator stage. I'm a bit leery of arbitrarily installing a package that isn't listed in the config. (see `imageconfigvalidator.go:validatePackages()` on how to enforce this). I see the need to prioritize installing the package early though, before the other packages have a chance to pollute the users.

Do we require shadow-utils for all user operations? I feel like there are cases were we don't want to install shadow-utils, but we still need to add users.

Reply:
nvm, `useradd` comes form `shadow-utils`. Still recommend adding this to the validator rather than arbitrarily installing it.

Reply:
Done. Thanks for your review!